provider/datagen/src/bin/icu4x-datagen.rs

```diff
@@ -0,0 +1,628 @@
+            "basic" => Some(CoverageLevel::Basic),
+            "moderate" => Some(CoverageLevel::Moderate),
+            "modern" => Some(CoverageLevel::Modern),
+            _ => None,
+        })
+        .collect::<Option<Vec<_>>>()
+    {
+        Some(PreprocessedLocales::LanguageIdentifiers(
+            provider
+                .locales_for_coverage_levels(locale_subsets.into_iter())?
+                .into_iter()
+                .collect(),
+        ))
+    } else {
+        if cli.locales.as_slice() == ["all"] {
+            log::warn!(
+                "`--locales all` selects the Allar language. Use `--locales full` for all locales"
+            );
+        }
+        None
+    };
+
+    if cli.without_fallback || matches!(cli.fallback, Fallback::Preresolved) {
+        driver = driver.with_locales_no_fallback(
+            match preprocessed_locales {
+                Some(PreprocessedLocales::All) => {
+                    eyre::bail!("--without-fallback needs an explicit locale list")
+                }
+                Some(PreprocessedLocales::LanguageIdentifiers(lids)) => lids,
+                None => cli
+                    .locales
+                    .into_iter()
+                    .map(|family| family.parse().wrap_err(family))
+                    .collect::<eyre::Result<Vec<_>>>()?,
+            },
+            Default::default(),
+        );
+    } else {
+        let locale_families = match preprocessed_locales {
+            Some(PreprocessedLocales::All) => vec![LocaleFamily::full()],
+            Some(PreprocessedLocales::LanguageIdentifiers(lids)) => lids
+                .into_iter()
+                .map(LocaleFamily::with_descendants)
+                .collect(),
+            None => cli
+                .locales
+                .into_iter()
+                .map(|family| family.parse().wrap_err(family))
+                .collect::<eyre::Result<Vec<_>>>()?,
+        };
+        let mut options: FallbackOptions = Default::default();
+        options.deduplication_strategy = match (
+            cli.deduplication_strategy,
+            cli.fallback,
+            cli.without_fallback,
+        ) {
+            (None, _, true) => None,
+            (Some(_), _, true) => {
+                eyre::bail!("cannot combine --without-fallback and --deduplication-strategy")
+            }
+            (Some(x), _, false) => match x {
+                DeduplicationStrategy::Maximal => Some(icu_datagen::DeduplicationStrategy::Maximal),
+                DeduplicationStrategy::None => Some(icu_datagen::DeduplicationStrategy::None),
+            },
+            (None, fallback_mode, false) => match fallback_mode {
+                Fallback::Auto => None,
+                Fallback::Hybrid => Some(icu_datagen::DeduplicationStrategy::None),
+                Fallback::Runtime => Some(icu_datagen::DeduplicationStrategy::None),
+                Fallback::RuntimeManual => Some(icu_datagen::DeduplicationStrategy::None),
```

Comment:
Oops, I think this came from earlier
```suggestion
                Fallback::Runtime => Some(icu_datagen::DeduplicationStrategy::Maximal),
                Fallback::RuntimeManual => Some(icu_datagen::DeduplicationStrategy::Maximal),
```